### PR TITLE
Some adjustment for Forms

### PR DESCRIPTION
**Pull request by NicolasDorier**
* Properly validate the parameters before setting them.
* Remove some dead code and `requiresRefundEmail` being duplicated in some situations.
* Refactor the Input form providers, so they can implement validation.
* Removing subclasses of `Field`, as we can't properly serialize/deserialize them.
* Ensure redirecturl is local for form builder (#4349)

### CODE REVIEW

File: BTCPayServer/Controllers/UIPaymentRequestController.cs [hunk 1]
```diff
@@ -195,32 +201,37 @@ public async Task<IActionResult> ViewPaymentRequestForm(string payReqId, [FromFo
...
                 case { } when string.IsNullOrEmpty(prFormId):
-                    break;
-                
+                case { } when Request.Method == "GET" && prBlob.FormResponse is not null:
+                    return RedirectToAction("ViewPaymentRequest", new { payReqId });
+                case { } when Request.Method == "GET" && prBlob.FormResponse is null:
+                        break;
                 default:
                     // POST case: Handle form submit
-                    if (!string.IsNullOrEmpty(formData) && formId == prFormId)
```
Comment (dennisreimann) [inline BTCPayServer/Controllers/UIPaymentRequestController.cs:202]: I think the `formId == prFormId` part should get applied to the `if` below.

File: BTCPayServer/Plugins/PointOfSale/Controllers/UIPointOfSaleController.cs [hunk 2]
```diff
@@ -230,9 +234,12 @@ public async Task<IActionResult> ViewPointOfSale(string appId, PosViewType? view
                 
                 default:
                     // POST case: Handle form submit
-                    if (!string.IsNullOrEmpty(formData) && formId == posFormId)
```
Comment (dennisreimann) [inline BTCPayServer/Plugins/PointOfSale/Controllers/UIPointOfSaleController.cs:233]: Same for `formId == posFormId` here.

File: BTCPayServer/Controllers/UIPaymentRequestController.cs (hunk 1 above)
Comment (NicolasDorier) [reply BTCPayServer/Controllers/UIPaymentRequestController.cs:202]: `formId` doesn't exists, found out we could remove it as the only form id that mater is the one in the blob.

**Comment by Kukks** [comment]
By making everything on field without the subclass, I don't know how we will be able extend later for more advanced scenarios.

**Comment by NicolasDorier** [comment]
Let's extend it later, we couldn't use it for now.

**Pull request merged**